### PR TITLE
Build tags: #5

### PR DESCRIPTION
**Pull request by acvetkov**
@vitalets 

https://github.com/acvetkov/teamcity-client/issues/5

### CODE REVIEW

File: src/tags/url.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * Convert tag list to JSON object
+ * @see https://confluence.jetbrains.com/display/TCD9/REST+API#RESTAPI-BuildTags
+ * @param {Array<String>|String} tags
+ * @returns {Object}
+ */
+export function tagsToObject(tags) {
+    assertTagsArgument(tags);
+    const arrTags = normalizeTags(tags);
+    return arrTags.reduce((result, tag) => {
+        result.tag.push({name: tag});
```
Comment (vitalets) [inline src/tags/url.js:28]: maybe better `result.tags` ?

File: src/tags/url.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * @param {Array<String>|String} tags
+ * @returns {String}
+ */
+export function serializeTags(tags) {
+    return JSON.stringify(tagsToObject(tags));
+}
+
+/**
+ * Assert tags arguments
+ * @param {*} tags
```
Comment (vitalets) [inline src/tags/url.js:45]: `{Array|String}`

File: src/utils/url.js [hunk 3]
```diff
@@ -28,3 +28,12 @@ export function locatorToString(data) {
...
     return _.compact(options).join();
 }
+
+/**
+ * Get query string
+ * @param {Object} data
+ * @returns {String}
+ */
+export function serializeParams(data) {
+    return _.map(data, (val, key) => `${key}=${val}`).join('&');
```
Comment (vitalets) [inline src/utils/url.js:38]: use uri.js! :)

**Comment by vitalets** [comment]
:+1:

File: src/tags/url.js (hunk 1 above)
Comment (acvetkov) [inline src/tags/url.js:28]: No, it's teamcity Api requirement.
We must post structure

``` json
{
  "count": 2,
  "tag": [
    {"name": "tag1"},
    {"name": "tag2"}
  ]
}
```

File: src/tags/url.js (hunk 2 above)
Comment (acvetkov) [inline src/tags/url.js:45]: It's assert function, and I suppose, that you can pass anything

File: src/utils/url.js (hunk 3 above)
Comment (acvetkov) [inline src/utils/url.js:38]: =)

**Pull request merged**